network-api/networkapi/wagtailpages/tests/blog/test_blog_index.py

```diff
@@ -305,6 +306,41 @@ def test_topic_route_shows_only_entries_of_topic(self):
         self.assertIn(topic_blog_page, response.context["entries"])
         self.assertNotIn(other_blog_page, response.context["entries"])
 
+    def test_index_intro_updated_with_topic_intro(self):
+        topic_intro_text = "This is a test topic intro."
+        topic = blog_factories.BlogPageTopicFactory(name="Test topic", intro=topic_intro_text)
+
+        url = self.get_topic_route(topic=topic.slug)
+
+        response = self.client.get(path=url)
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertEqual(response.context["index_intro"], topic_intro_text)
+
+    def test_index_title_updated_with_topic_title(self):
+        topic_title = "Test Topic Title"
+        topic = blog_factories.BlogPageTopicFactory(name="Test topic", title=topic_title)
+
+        url = self.get_topic_route(topic=topic.slug)
+
+        response = self.client.get(path=url)
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertEqual(response.context["index_title"], topic_title)
+
+    def test_index_title_defaults_to_topic_name_if_no_title(self):
+        topic_name = "Test Topic Name"
+        # Expecting the name of the topic -> the name of the index page
+        expected_index_title = "Test Topic Name Blog"
+        topic = blog_factories.BlogPageTopicFactory(name=topic_name, title="")
+
+        url = self.get_topic_route(topic=topic.slug)
+
+        response = self.client.get(path=url)
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertEqual(response.context["index_title"], expected_index_title)
```

Comment:
Hi @danielfmiranda , I'm not familiar with how these tests are usually written. Do you mind explaining where `expected_index_title` is getting set? I see it was defined on line 334 but can't' figure out how the value is used. Is it because the factory always generates a blog index page with title "Blog"? If so, I'm wondering why `expected_index_title` wouldn't just be a single word "Blog".

Reply:
Hi @mmmavis! No problem. 

Blog Topic detail pages such as https://foundation.mozilla.org/en/blog/topic/advocacy/ are technically still the `BlogIndexPage`, just filtered for only blog posts that are linked to that following topic.

In the blog index page's [`filter_entries_for_topic` function,](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/f9a4c7ca64801437e40413dcb480b653f4910861/network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py#L177C1-L177C58), we have the following lines:


```python
...
        locale = get_locale_from_request(context["request"])
        try:
            localized_topic = topic.get_translation(locale)
        except ObjectDoesNotExist:
            localized_topic = topic

        context["index_intro"] = localized_topic.intro
        context["index_title"] = titlecase(f"{localized_topic.name} {self.title}")

        if localized_topic.title:
            context["index_title"] = localized_topic.title
...
```

What this code is pretty much saying, in regards to the index title, is "if the topic has a custom `title` set, update `index_title` accordingly."

If the topic does **not** have a title set, have `index_title` default to the value of `<topic_name> <index_page_title>`.  Which in my previous commit's case would be "Blog".

However, taking a second look at that approach, I can see that its quite confusing.

So instead, I have updated/renamed the function in my latest commit, as well as adding an additional comment in hopes it makes it easier to understand.

```python
    def test_index_title_default_works_with_no_title(self):
        topic = blog_factories.BlogPageTopicFactory(name="Test Topic", title="")

        # If a topic has no title set, the blog index page will set the context's
        # "index_title" field to the default value of "<topic_name> <blog_index_page_title>"
        expected_index_title_value = f"{topic.name} {self.blog_index.title}"

        url = self.get_topic_route(topic=topic.slug)

        response = self.client.get(path=url)

        self.assertEqual(response.status_code, HTTPStatus.OK)
        self.assertEqual(response.context["index_title"], expected_index_title_value)

```

Do you think you could please take a look whenever you find a free moment and let me know what you think?

Thanks!